src/HyperlaneArbiter.sol

```diff
@@ -124,13 +124,59 @@ contract HyperlaneArbiter is Router {
         );
     }
 
+    /**
+     * @notice Quotes a compact intent and returns the msg.value that should be provided
+     * cover all hyperlane fees (relay, etc).
+     * @param claimChain The chain ID of the claim.
+     * @param compact The compact intent to fill.
+     * @dev signatures must be compliant with https://eips.ethereum.org/EIPS/eip-2098
+     * @param allocatorSignature The allocator's signature.
+     * @param sponsorSignature The sponsor's signature.
+     * @return The quoted fee.
+     */
+    function quote(
+        uint32 claimChain,
+        Compact calldata compact,
+        Intent calldata intent,
+        bytes calldata allocatorSignature,
+        bytes calldata sponsorSignature
+    ) external view returns (uint256) {
+        require(block.chainid == intent.chainId, "invalid chain");
+
+        // TODO: support Permit2 fills
+        address filler = msg.sender;
+
+        return _Router_quoteDispatch(
+            claimChain,
+            Message.encode(compact, allocatorSignature, sponsorSignature, hash(intent), intent.fee, filler),
+            "",
```

Comment:
if we dont populate anything here, the `handle` function is assumed to have a gas limit of 50k
we may need to adjust this, have you successfully used these quotes in prod yet?

Reply:
not yet — open question on whether to just build in an approximate buffer and let filler work out the exact quoted fee or if the UI should derive it locally